src/common/Json/JsonHelpers.cs

```diff
@@ -10,100 +10,141 @@
-
-        public static string SerializeDictionary(IDictionary<string, string> d)
-            => SerializeObject(d);
-
-        public static string SerializeDictionaryIndented(IDictionary<string, string> d)
-            => SerializeObjectIndented(d);
-
-        public static dynamic ParseJson(string s)
-            => (dynamic)JsonNode.Parse(s);
+        /// <summary>
+        /// Serialize an object to JSON. PII values are output unscrambled. Swallows serialization exceptions.
+        /// </summary>
+        /// <param name="input">Object or value to serialize</param>
+        /// <returns>String representation of <paramref name="input"/></returns>
+        public static string SerializeForLoggingPurposeIndented(object input)
+        {
+            return SerializeForLoggingPurpose(input, SerializerOptionsForLoggingPurposeIndented);
+        }
 
-        public static T ParseAndGetProperty<T>(string json, string property)
+        /// <summary>
+        /// Serialize an object to JSON. PII values are output unscrambled. Swallows serialization exceptions.
+        /// </summary>
+        /// <param name="input">Object or value to serialize</param>
+        /// <returns>String representation of <paramref name="input"/></returns>
+        public static string SerializeForLoggingPurpose(object input)
         {
-            var obj = JsonNode.Parse(json);
-            return obj[property].GetValue<T>();
+            return SerializeForLoggingPurpose(input, SerializerOptionsForLoggingPurpose);
         }
 
-        public static JsonSerializerOptions CreateSerializerOptions(bool indented = false, bool camelCaseContextResolver = true, bool caseInsensitive = false)
+        private static string SerializeForLoggingPurpose(object input, JsonSerializerOptions settings)
         {
-            var jsonSerializerOptions = new JsonSerializerOptions
+            if (input is string inputString)
             {
-                PropertyNamingPolicy = camelCaseContextResolver ? JsonNamingPolicy.CamelCase : null,
-                WriteIndented = indented,
-                DefaultIgnoreCondition = JsonIgnoreCondition.Never,
-                PropertyNameCaseInsensitive = caseInsensitive,
+                return inputString;
+            }
 
-                ReferenceHandler = ReferenceHandler.IgnoreCycles,
-            };
+            if (input is PII pii)
+            {
+                return pii.Value;
+            }
 
-            foreach (var converter in DefaultConverters)
+            if (input is Exception ex)
             {
-                jsonSerializerOptions.Converters.Add(converter);
+                input = RemoveUnwantedExceptionProperties(input, ex);
             }
 
-            return jsonSerializerOptions;
+            try
+            {
+                return JsonSerializer.Serialize(input, settings);
+            }
+            catch (Exception)
+            {
+                return "Serialization Error";
```

Comment:
@Eneuman I have seen this happening quite a few times in the logs, is there any way we can avoid this or improvise this with ExceptionConverters, right now exception converters does only remove ```TargetSite.Module```. Suggestions or thoughts please ?

Reply:
@hsubramanianaks Yeah, I have seen this aswell. I think the main problem why theese are written to the log is because of how CancelationExceptions are handled. Today they are handled as regular exceptions and hence written to the log. The problem is that CancelationException cannot be serialized and thats why "Serialization Error" occurs.
After this PR, "Serialization Error" can only occure when writing objects to the logsfile, so they can safely be ignored until CancelationException handling is refactored.

I could change the output to "Serialization Error of type: {input.GetType()} when trying to write to logfile." to make it more precise?

Reply:
@Eneuman Yes that would help a lot. Apologies for late reply.

Reply:
@hsubramanianaks Fixed!